templates/default/resource.rb.erb

```diff
@@ -1 +1,8 @@
-# To learn more about Custom Resources, see https://docs.chef.io/custom_resources/
+resource_name :<%= cookbook_name.gsub('-', '_') %>_<%= recipe_name %>
```

Comment:
You need to also add ``provides`` here as it's a requirement for Chef 16. If you use the latest cookstyle, I bet you this generated file will fail.